export/export.go

```diff
@@ -0,0 +1,432 @@
+	}
+	if err := errgrp.listPvcs(ctx, &s.pvcService, &resources); err != nil {
+		return nil, xerrors.Errorf("call listPvcs: %w", err)
+	}
+	if err := errgrp.listStorageClasses(ctx, &s.storageClassService, &resources); err != nil {
+		return nil, xerrors.Errorf("call listStorageClasses: %w", err)
+	}
+	if err := errgrp.listPcs(ctx, &s.priorityclassService, &resources); err != nil {
+		return nil, xerrors.Errorf("call listPcs: %w", err)
+	}
+	if err := errgrp.getSchedulerConfig(ctx, &s.schedulerService, &resources); err != nil {
+		return nil, xerrors.Errorf("call getSchedulerConfig: %w", err)
+	}
+
+	if err := errgrp.Grp.Wait(); err != nil {
+		return nil, xerrors.Errorf("get resources all: %w", err)
+	}
+	return &resources, nil
+}
+
+func (s *Service) Export(ctx context.Context) (*ResourcesForExport, error) {
+	resources, err := s.get(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("export resources all: %w", err)
+	}
+	return resources, nil
+}
+
+// Import all resources from posted data.
+// (1) Restart scheduler based on the data.
+// (2) Apply each resource to the scheduler.
+//     * If UID is not nil, an error will occur. (try to find existing resource by UID)
+// (3) Get all resources. (Separated the get function to unify the struct format.)
+func (s *Service) Import(ctx context.Context, resources *ResourcesForImport) error {
+	if err := s.schedulerService.RestartScheduler(resources.SchedulerConfig); err != nil {
+		return xerrors.Errorf("restart scheduler with imported configuration: %w", err)
+	}
+	errgrp := NewErrGroupWithSemaphore(ctx)
+
+	if err := errgrp.applyPcs(ctx, &s.priorityclassService, resources); err != nil {
+		return xerrors.Errorf("call applyPcs: %w", err)
+	}
+	if err := errgrp.applyStorageClasses(ctx, &s.storageClassService, resources); err != nil {
+		return xerrors.Errorf("call applyStorageClasses: %w", err)
+	}
+	if err := errgrp.applyPvcs(ctx, &s.pvcService, resources); err != nil {
+		return xerrors.Errorf("call applyPvcs: %w", err)
+	}
+	if err := errgrp.applyNodes(ctx, &s.nodeService, resources); err != nil {
+		return xerrors.Errorf("call applyNodes: %w", err)
+	}
+	if err := errgrp.applyPods(ctx, &s.podService, resources); err != nil {
+		return xerrors.Errorf("call applyPods: %w", err)
+	}
+
+	if err := errgrp.Grp.Wait(); err != nil {
+		return xerrors.Errorf("apply each resources: %w", err)
+	}
+
+	// The apply of PersistentVolume will be called after the apply of PersistentVolumeClaim is finish.
+	// This is because that PersistentVolume search PersistentVolumeClaim and associate itself if they had relationship.
+	if err := errgrp.applyPvs(ctx, &s.pvService, &s.pvcService, resources); err != nil {
+		return xerrors.Errorf("call applyPvs: %w", err)
+	}
+	if err := errgrp.Grp.Wait(); err != nil {
+		return xerrors.Errorf("apply applyPvs: %w", err)
+	}
+
+	return nil
+}
+
+func (eg *ErrGroupWithSemaphore) listPods(ctx context.Context, s *PodService, r *ResourcesForExport) error {
```

Comment:
All `listHoge` and `applyHoge` receive HogeService as pointer. It isn't needed to be pointer. 
With this change, we can also change `(*s).List(ctx)` in L205 to `s.List(ctx)`.

```suggestion
func (eg *ErrGroupWithSemaphore) listPods(ctx context.Context, s PodService, r *ResourcesForExport) error {
```

Reply:
> With this change, we can also change (*s).List(ctx) in L205 to s.List(ctx).

Due to the last fix, I will use `Service` as method receiver.